### ISSUE TITLE
Feature Idea - append Retry count and pass rate to the report?

### ISSUE THREAD

**Issue by szymon-kazmierczak** [lyndsey-ferguson/fastlane-plugin-test_center#43]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the example [Fastfile](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/fastlane/Fastfile)

### Issue Description
I'm using multi_scan, great work on it btw :)
Is it possible to add the retry count for each test (if a test was rerun), and the pass rate of it to the report (both junit and html)? My goal is to have a meaningful HTML report for viewing on jenkins, and to store the metrics in Datadog, for which we're planning to use the junit/xml format tests from both android and iOS.

**Comment by lyndsey-ferguson** [comment]
Thanks @szymon-kazmierczak, it is possible to add that data. While you are waiting for this feature, you _could_ pass a `:testrun_completed_block` to `multi_scan` which is called after each run (successful or not) to collect this data and add it to your report file. Let me know if that would help you (and I could provide guidance if you like).

**Comment by szymon-kazmierczak** [comment]
Sounds good, I'll use the completed block in the meantime. Thanks!

**Comment by lyndsey-ferguson** [comment]
@szymon-kazmierczak what does it mean for a test to have a "pass rate"? I can understand adding an attribute to each test "retry_count" being the count of how many times we have run that (test - 1).

**Comment by lyndsey-ferguson** [comment]
Looking at this, I can leverage the "junit_helper" to parse the file, update the attributes for retried tests, and then save it again.

**Comment by szymon-kazmierczak** [comment]
Now that I think about it, pass rate would be how many times a test passed in the whole test run. 
So if it took 3 tries to pass a test, the pass rate would be 1/4 (25%) - does that make sense? Basically verifying the flakiness of a test - some could fail once in a long time, some could be super flaky and only passing once in a while - my goal is to figure out the stats for that.

**Comment by lyndsey-ferguson** [comment]
Can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane to review the test results?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-43-retrycount-passrate-in-junit"
```

Please attach the console output as a text file to this issue (makes it easier for me to review).

**Comment by lyndsey-ferguson** [comment]
I just changed the code to do the retry count update in the `collate_junit_reports` action where it is better suited. If you have already updated your `Pluginfile`, please delete your `Gemfile.lock` file, rerun `bundle install` and then run your fastlane.

**Comment by lyndsey-ferguson** [comment]
@szymon-kazmierczak would you be able to run your fastlane to review the test results in order to see if it fits your needs? I did not add pass rate.

**Comment by szymon-kazmierczak** [comment]
Sorry, haven't been able to run this yet as my setup changed. I'll recreate it sometime this week and report back :)

**Comment by lyndsey-ferguson** [comment]
Addressed in [v3.6.0](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.6.0).
